### ISSUE TITLE
Research where to store data for returning flights

### ISSUE THREAD

**Issue by olgashi** [cupofsugarCo/flying_penguin#72]

**Comment by olgashi** [comment]
@bsotiroff 
I thought about this for a little bit. I think this is kind of related to couple of other issues we discussed (timing out issue and splitting up the requests for departure and return flights).

I think what we actually need to do is to create an offer request and set `return_offers` to `false`, so that we don't get any offer data in the response. 

But then just send another request (`List Offers`), where we could actually use pagination, sorting, and filtering (whatever makes sense), where we limit results to 10 offers (or whatever the acceptable number is) and just departing flights. 

Then when user picks a departing flight it should trigger a request to get the returning flights from the offer that we have previously created.

We can experiment with this approach and see if the response times are better overall and if we run into any of the timeout issues.

With this I think we don't need to store any data anywhere (the return flight data). I don't think there is a point, since we can go and get it on demand.

In any case we can try this approach and reevaluate if we find any issues with it.

**Comment by olgashi** [comment]
@bsotiroff 
This talk is pretty good. Checj it out when you have time. Its on when to use databases in phoenix/elixir (starts at 19:11)
https://www.youtube.com/watch?v=MTT1Jl4Fs-E&t=1151s

**Comment by bsotiroff** [comment]
> I think what we actually need to do is to create an offer request and set return_offers to false, so that we don't get any offer data in the response.
> But then just send another request (List Offers), where we could actually use pagination, sorting, and filtering (whatever makes sense), where we limit results to 10 offers (or whatever the acceptable number is) and just departing flights.

depending on the info that we get with a return_offers=false response. I think this is a really promising strategy.  If we can get cost and number of stops for example. Then we would have enough to sort and paginate.

In regards to the elixirconf talk, it makes sense that we can just keep a state for a user until they actually book a flight. How we do that, I'm not sure. It might be enough to have "channel state" and not necessarily a GenServer, but as we learn more about the react integration with phoenix Channels, It should become clearer. I read up on it a little bit yesterday, but I haven't been able to play with the examples online in order to get familiar with it.

Furthermore. I did look into a strategy on how to pre-fetch the return flight data. I came across [this forum](https://elixirforum.com/t/how-to-make-async-requests-using-httpoison/19349/12) and elixir has this concept of Task.start and await. It seems similar to Promises in javascript, although I think this creates an entirely new process. I was thinking we could leverage this in order to begin fetching the results immediately after the departure flights have been returned. That way, we are still using the time the user is browsing for departures.

In terms of "storage" I think this would end up being stored in either the gen_server or phoenix channel state (wherever we decide to hold) rather than the db since it is ephemeral and doesn't need to be persisted until the user books a flight.

**Comment by bsotiroff** [comment]
you're totally right about how offers work, I'm going to create a new issue / PR to break it into an offer request and offer list. Regardless of performance gains, this is how to implement pagination. Although I imagine there will likely be some performance gains as well.
